### PR TITLE
Update Plugin List - Explicitly deprecate out-of-date plugins

### LINKED ISSUE #3728: Explain the tags meaning for plugins file
Users are confused

Screen Shot 2021-03-22 at 12 55 41 PM

### LINKED ISSUE #5073: Change Link Of Recommended Visual Testing Plugin
### Subject

Visual Testing

### Description

The guide recommends a plugin that is not currently compatible with Cypress and the repo appears to be unmaintained 
https://github.com/meinaart/cypress-plugin-snapshots

Can we get a better recommendation for setup of visual testing? There are quite a few open source ones but I can image others getting frustrated with a link to a plugin that doesn't work

### LINKED ISSUE #5086: Remove mentions to outdated plugin: cypress-plugin-snapshots
In [this](https://docs.cypress.io/guides/tooling/visual-testing) and [this](https://docs.cypress.io/plugins#Visual%20Testing) pages there are mentions and instructions to install a plugin that has not worked with Cypress since version 4.5.0: `cypress-plugin-snapshots`.

The first page I mentioned even has instructions to install this plugin:

```
  // run 'npm i cypress-plugin-snapshots -S'
  // capture the element screenshot and
  // compare to the baseline image
```

Which if attempted with any recent version of Cypress (> 2020) will give this error:
```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: my-project@1.0.0
npm ERR! Found: cypress@12.6.0
npm ERR! node_modules/cypress
npm ERR!   dev cypress@"^12.6.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer cypress@"^4.5.0" from cypress-plugin-snapshots@1.4.4
npm ERR! node_modules/cypress-plugin-snapshots
npm ERR!   cypress-plugin-snapshots@"*" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

Because of [this line](https://github.com/meinaart/cypress-plugin-snapshots/blob/master/package.json#L61) in that plugin's package.json:

```json
  "peerDependencies": {
    "cypress": "^4.5.0"
  },
```

I think there is no purpose on having instructions for a command that won't work if attempted. This Issue is to ask if it can be removed so the next person can focus on visual testing plugins that are maintained.

### LINKED ISSUE #5293: Old plugins and use of Legacy section
### Subject

Contributing - Plugins

### Description

The [Plugins list](https://docs.cypress.io/plugins) includes a link "Follow these instructions to [submit your own plugin](https://github.com/cypress-io/cypress-documentation/blob/master/CONTRIBUTING.md#adding-plugins)." as a section in the [CONTRIBUTING](https://github.com/cypress-io/cypress-documentation/blob/master/CONTRIBUTING.md) document - see below under "Adding Plugins" for full text.

It however does not deal with the situation that a listed plugin no longer meets the criteria for new submission.

## Suggestion

Add a new section to [CONTRIBUTING](https://github.com/cypress-io/cypress-documentation/blob/master/CONTRIBUTING.md) which deals with maintaining plugins.

- What are the criteria for plugins to remain on the [Plugins list](https://docs.cypress.io/plugins)?
- What are the criteria for a plugin to be listed in the [Plugins list > Legacy section](https://docs.cypress.io/plugins#Legacy)? For instance is there a minimum version of Cypress which the plugin needs to be compatible with?

## Discussion

The [Plugins list > Legacy](https://docs.cypress.io/plugins#Legacy) section does not define the meaning of "Legacy", so unless there is some text added to the Legacy section, it is unclear what it means when a plugin is listed in this section.

- Legacy could refer to a plugin which runs only on Cypress 9 and below using a [Configuration (Legacy)](https://docs.cypress.io/guides/references/legacy-configuration)
- Legacy could also refer to plugins which are archived, deprecated or otherwise unsupported

## Problematic plugin states

- Plugin's npm registry entry is deprecated
- Plugin is explicitly deprecated in source repo
- Plugin's source repo is archived
- Plugin is not compatible with current major version of Cypress
- Plugin's CI pipeline fails

### @cypress/xpath

- npm package [@cypress/xpath](https://www.npmjs.com/package/@cypress/xpath)

```text
$ npm view @cypress/xpath deprecated
Package no longer supported. Contact Support at https://www.npmjs.com/support for more info.
```

- Source https://github.com/cypress-io/cypress/tree/develop/npm/xpath#readme

> @cypress/xpath has been deprecated and is no longer supported.

### @cypress/skip-test

- npm package [@cypress/skip-test](https://www.npmjs.com/package/@cypress/skip-test)

- Source https://github.com/cypress-io/cypress-skip-test archived on Jan 31, 2023.

- https://github.com/cypress-io/cypress-skip-test/blob/1f9bb3ee95dcbc3884c4cc2c8d1692509ba0ab44/package.json#L23 specifies old `"cypress": "6.2.1"`

- CircleCI https://app.circleci.com/pipelines/github/cypress-io/cypress-skip-test no longer configured

- Last commit https://github.com/cypress-io/cypress-skip-test/commit/1f9bb3ee95dcbc3884c4cc2c8d1692509ba0ab44 failed CircleCI

---
## CONTRIBUTING
Text copied from https://github.com/cypress-io/cypress-documentation/blob/master/CONTRIBUTING.md#adding-plugins:

### Adding Plugins

To add a plugin, submit a [pull request](https://github.com/cypress-io/cypress-documentation/blob/main/CONTRIBUTING.md#pull-requests) with the corresponding data added to the [`plugins.json`](https://github.com/cypress-io/cypress-documentation/blob/main/src/data/plugins.json) file. Your plugin should have a name, description, link to the plugin's code, as well as any keywords.

We want to showcase plugins that work and have a good developer experience. This means that a good plugin generally has:

1. Purpose of plugin articulated up front
1. Installation guide
1. Options and API are documented
1. Easy to follow documentation. Users should not have to read the source code to get things working.

Each plugin submitted to the plugins list should have the following:

1. Integration tests with Cypress

   - Demonstrates the plugin working
   - Acts as real-world example usage

2. CI pipeline
3. Compatibility with at least the latest major version of Cypress

Plugins are listed in in the following order to for users:

- official
- verified
- community
- experimental
---

### PR DESCRIPTION
**Pull request by dingraham**
The Plugins List currently contains a ton of unsupported or archived submissions. 
This change cleans up the list and explicitly labels these plugins as _deprecated_.

This was done by 
- Removing the `experimental` badge and renaming those select few plugins with `community`. In my opinion community plugins are "experimental" by nature if they aren't specifically `verified` by the Cypress team so this felt duplicated. Especially because only a handful had this distinction.
- Adding a few descriptors to what each badge represents to the best of my knowledge
- Adding the `deprecated` badge for plugins that meet **at least one of these three** following conditions
1. Npm registry deprecated
2. Source Repo Publicly Archived
3. Incompatible with v10+ (determined by inactivity over 3+ years ago and/or explicit community issues in the repo referencing incompatibility)
- Removing the Legacy section at the bottom and moving those three plugins into their respective categories as deprecated

After, the giant list of plugins was reviewed, one by one, and those considered "deprecated" were moved to the end of the list for that category. In a few cases I removed the plugin entirely, especially if the plugin was deprecated and clearly unused by the community. I've broken down why each plugin was deprecated below. 

If this change is too large I can break this list of changes up into multiple PRs for each section. There were a total of 47 plugins updated, which reflects the need for updating this list 😄 

**LIST OF DEPRECATED PLUGINS**

**Preprocessors:**
- [cytorus](https://github.com/NaturalIntelligence/cytorus) (v10+ incompatible)
- [cypress-app-watcher-preprocessor](https://github.com/TheBrainFamily/cypress-app-watcher-preprocessor) (v10+ incompatible, 5 years ago)
- [cypress-eslint-preprocessor](https://github.com/chinchiheather/cypress-eslint-preprocessor) (v10+ incompatible, 6 years ago)
- [cypress-rollup](https://github.com/bahmutov/cy-rollup) (v10+ incompatible)
- [cypress-esbuild-preprocessor](https://github.com/sod/cypress-esbuild-preprocessor)  (unsure what version is support, 4+ years ago, @bahmutov/cypress-esbuild-preprocessor plugin does the same thing and is maintained)

**Development Tools:**
- [@cypress/instrument-cra](https://github.com/cypress-io/instrument-cra) (publicly archived)
- [cypress-dark](https://github.com/bahmutov/cypress-dark) (super popular but v10+ incompatible, would like to replace in a follow-up PR with my plugin cypress-runner-themes which is v10+ supported)
- [cypress-protobuf](https://github.com/NoriSte/cypress-protobuf) (v10+ incompatible, 4 years ago) 
- [knapsack-pro-cypress](https://github.com/KnapsackPro/knapsack-pro-cypress) (publicly archived)
- [cypress-autostub](https://github.com/dan-cooke/cypress-autostub) (v10+ incompatible, 4 year ago)
- [cypress-plugin-liverload](https://github.com/unlocomqx/cypress-plugin-livereload) (removed entirely - no downloads on npm, v10+ incompatible, 4 years ago)
- [cypress-select-tests ](https://github.com/bahmutov/cypress-select-tests) (publicly archived)
- [cypress-autorecord](https://github.com/Nanciee/cypress-autorecord) (v10+ incompatible, 3 years ago)
- [cypress-fiddle](https://github.com/cypress-io/cypress-fiddle) (publicly archived)
- [npm-cy](https://github.com/bartlett705/npm-cy) (removed entirely - no npm repo, 5+ years ago)
- [cypress-extends](https://github.com/bahmutov/cypress-extends) (v10+ incompatible)
- [Browserify](https://github.com/cypress-io/cypress-browserify-preprocessor) (moved from Legacy)

**Custom Commands:**
- [cypress-unfetch](https://github.com/RcKeller/cypress-unfetch) (v10+ incompatible, 5 years ago)
- [cypress-redux](https://github.com/RcKeller/cypress-redux) (v10+ incompatible, 5 years ago)
- [cypress-graphql-mock](https://github.com/tgriesser/cypress-graphql-mock) (removed entirely - no npm repo, v10+ incompatible, 5 years ago, other updated solutions in list)
- [cypress-pipe](https://github.com/NicholasBoll/cypress-pipe) (v10+ incompatible, 4 years ago)
- [cypress-plugin-stripe-elements](https://github.com/dbalatero/cypress-plugin-stripe-elements) (v10+ incompatible, 3 years ago)
- [cypressautomocker](https://github.com/scottschafer/cypressautomocker) (removed entirely - no npm downloads, v10+ incompatible, 5 years ago)
- [cypress-skip-test](https://github.com/cypress-io/cypress-skip-test) (publicly archived)
- [cypress-websocket-testing](https://github.com/lensesio/cypress-websocket-testing)  (v10+ incompatible, 5 years ago)
- [cypress-iframe](https://gitlab.com/kgroat/cypress-iframe)  (v10+ incompatible, 4 years ago)
- [cypress-rest-graphql](https://github.com/BrandedEntertainmentNetwork/cypress-rest-graphql) (v10+ incompatible, 3 years ago)
- [cypress-forced-colors](https://github.com/ling1726/cypress-forced-color) (removed entirely - no npm downloads, v10+ incompatible, 3 years ago)
- [cypress-dotenv](https://github.com/morficus/cypress-dotenv) (moved from Legacy)
- [cypress-xpath](https://github.com/cypress-io/cypress/tree/develop/npm/xpath) (moved from Legacy)

**Extending other testing frameworks:**
- [cyphell](https://github.com/intuit/cyphfell) (removed entirely - no npm registry, 6 years ago)
- [cypress-capybara](https://github.com/testdouble/cypress-capybara) (publicly archived)
- [cypress-jest-adapter](https://github.com/phongnd39/cypress-jest-adapter) (v10+ incompatible, 5 years ago)

**Authentication:** 
- [cypress-keycloak-commands](https://github.com/Fredx87/cypress-keycloak-commands) (verified but clear issues against v10+, no support for 4 years despite popularity)
- [cypress-otp](https://github.com/NoriSte/cypress-otp) (super popular but clear issues against v10+, no support for 5 years)

**Framework Tooling:**
- [Vue Client](https://github.com/vuejs/vue-cli/tree/dev/packages/%40vue/cli-plugin-e2e-cypress) (super popular but clear issues against v10+) 
- [cypress-nuxt](https://github.com/NickBolles/cypress-nuxt) (v10+ incompatible, 4 years ago, moved to component category since it's for component testing)
- [elm-batteries](https://github.com/cedricss/elm-batteries) (v10+ incompatible, 5 years ago)
- [cypress-laravel](https://github.com/noeldemartin/cypress-laravel) (v10+ incompatible, 5 years ago)
- [cypress-splitio](https://gitlab.com/davidggevorgyan/cypress-splitio) (v10+ incompatible, 3 years ago)
- [Bison](https://github.com/echobind/bisonapp) (removed entirely - removed cypress from their examples in lieu of playwright)

**Component Testing:**
- [cypress-angularjs-unit-test](https://github.com/bahmutov/cypress-angularjs-unit-test) (assuming v10+ incompatible, 7 years ago) 
- [cypress-cycle-unit-test](https://github.com/bahmutov/cypress-cycle-unit-test) (assuming v10+ incompatible, 6 years ago) 
- [cypress-hyperapp-unit-test](https://github.com/bahmutov/cypress-hyperapp-unit-test) (assuming v10+ incompatible, 6 years ago) 
- [cypress-svelte-unit-test](https://github.com/bahmutov/cypress-svelte-unit-test) (v10+ incompatible) 

**Visual Testing:**
- SmartBear VisualTest - fixed link
- [cypress-plugin-snapshots](https://github.com/meinaart/cypress-plugin-snapshots) (v10+ incompatible)
- [cypress-blink-test](https://github.com/PatrickWalker/cypress-visual-regression) (removed entirely - no npm downloads, 5 years ago)f

**Reporting:**
- [cypress-log-to-output](https://github.com/flotwig/cypress-log-to-output) (v10+ incompatible)
- [cypress-slack-healthcheck](https://github.com/bdimitrovski/cypress-healthcheck) (removed entirely - no npm registry)
- [cypress-msteams-reporter](https://github.com/maritome/cypress-msteams-reporter) (outdated, last update 3 years ago, Cypress has Teams support now)
- Moved a few similar plugins adjacent in the list

Example: 
image

Additionally, this addresses the following PRs:
- https://github.com/cypress-io/cypress-documentation/pull/4685
- https://github.com/cypress-io/cypress-documentation/pull/4744

Issues:
closes #5293
closes #5086 
closes #5073
closes #3728

### CODE REVIEW

**Comment by dingraham** [comment]
@jennifer-shehane, I'm confused why the CLA I signed isn't being recognized as "signed"
Also not sure why my email isn't being registered, would closing and reopening a new PR be the best approach to resolve this? Thank you!

**Comment by MikeMcC399** [comment]
Hi David (@dingraham)

The CLA error message doesn't give the full details, but what it is trying to tell you is that your commits are signed with `David Ingraham <davidingraham@Davids-MacBook-Pro.local>` and the e-mail address `davidingraham@Davids-MacBook-Pro.local` is not listed in any GitHub user's account.

> **David Ingraham** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).
> You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/cypress-io/cypress-documentation?pullRequest=5789) it.

Check your GitHub Settings > Emails.
Depending on how you have set up your account, you may have an e-mail address like `66998419+MikeMcC399@users.noreply.github.com` allocated to you.

Set up your Git email accordingly. For me that would be:

```shell
git config --global user.email "66998419+MikeMcC399@users.noreply.github.com"
```

You don't need to open a new PR to correct things. I would squash your commits (which should use any new e-mail address you set up) and force push the branch.

Here is the log showing your three commits followed by one from me:

```text
$ git log  -4 --format=email
From 00d6d46c6dd6518ab0a17cdc288c6d2272a74b07 Mon Sep 17 00:00:00 2001
From: David Ingraham <davidingraham@Davids-MacBook-Pro.local>
Date: Sun, 7 Apr 2024 12:43:48 -0600
Subject: [PATCH] update plugin url and plugind escription


From 0f9967953e526525d58f1ba2bdc93f762a2ae83e Mon Sep 17 00:00:00 2001
From: David Ingraham <davidingraham@Davids-MacBook-Pro.local>
Date: Sun, 7 Apr 2024 12:34:36 -0600
Subject: [PATCH] fix typoe


From 124b17aec58af2b5389ae1cf9ba569ce1a4bdada Mon Sep 17 00:00:00 2001
From: David Ingraham <davidingraham@Davids-MacBook-Pro.local>
Date: Sun, 7 Apr 2024 11:29:43 -0600
Subject: [PATCH] update plugin list with deprecated plugins


From 84a5a93276a6e39ebdbae86dd8151a6cba9eca1a Mon Sep 17 00:00:00 2001
From: Mike McCready <66998419+MikeMcC399@users.noreply.github.com>
Date: Thu, 4 Apr 2024 18:51:07 +0200
Subject: [PATCH] Fix miscellaneous typos (#5784)
```

**Comment by MikeMcC399** [comment]
@dingraham

Well done for bringing the list forward! I have not checked the whole list. I just did a couple of spot checks that looked fine.

File: src/components/plugins-list/style.module.css [hunk 1]
```diff
@@ -97,6 +89,14 @@ ul.pluginsList li p {
   background-color: #00805b;
 }
 
+.badgePill.badgeCommunity {
+  background-color: #737373;
+}
+
+.badgePill.badgeDeprecated {
+  background-color: #870b2a;
+}
```
Comment (MikeMcC399) [inline src/components/plugins-list/style.module.css:98]: I'm not the styling expert, but I wonder if the Community background could be made less bland and the Deprecated background less prominent?

This has to work for both light and dark themes.

Comment (dingraham) [reply src/components/plugins-list/style.module.css]: Just pushed up another change, thoughts on these instead?

image

**Review by MikeMcC399 — APPROVED** [review]
LGTM!

**Comment by MikeMcC399** [comment]
@dingraham 

To fix the CLA issue, you would need to manipulate the previous commits so `davidingraham@Davids-MacBook-Pro.local` isn't used anymore. The simplest way would be to squash and force push. Feel free to DM me if that's not clear.

**Review by jennifer-shehane — APPROVED** [review]
Thanks, this makes the plugins much clearer to sort through.

**Pull request merged**